lib/active_record/mysql/enum/mysql_adapter.rb

```diff
@@ -45,20 +45,30 @@ def type_to_sql(type, limit: nil, **_options) # :nodoc:
           end
         end
 
+        if Gem::Version.new(Rails.version.to_f.to_s) < Gem::Version.new('7.0')
+          private
 
-        private
+          def initialize_type_map(m = type_map)
+            super
+
+            m.register_type(/enum/i) do |sql_type|
+              limit = sql_type.sub(/^enum\('(.+)'\)/i, '\1').split("','").map { |v| v.to_sym }
+              ActiveRecord::Type::Enum.new(limit: limit)
+            end
```

Comment:
Wow, I had to do a little digging into the Rails 7 changes to understand why the below additions were necessary. They've removed all flexibility of new types being bolted onto the adapters by loading the mapping at abstract load time. Thank you for taking care of this and adding the support. Could we DRY up the `register_type` chunk that I've highlighted in this comment into its own helper method? That way we have one way of registering the enum type.